packages/mdc-slider/index.js

```diff
@@ -162,6 +165,44 @@ class MDCSlider extends MDCComponent {
     this.max = parseFloat(this.thumb_.getAttribute(strings.ARIA_VALUEMAX)) || this.max;
     this.step = parseFloat(this.thumb_.getAttribute(strings.DATA_STEP)) || this.step;
     this.value = origValueNow || this.value;
+    if (this.tickMarkSet_) {
+      this.setUpTickMarks();
+    }
+  }
+
+  setUpTickMarks() {
```

Comment:
There's a lot of logic here that could be moved to the foundation:
```
setUpTickMarks() {
    const numMarks = this.foundation_.calculateNumberOfTickMarks();

    // Remove tick marks if there are any
    ...
    // Create the tick marks and append to the tick mark set
    ...

    this.foundation_.adjustLastTickMark(numMarks);
}
```
For `foundation.adjustLastTickMark()`, I would add an adapter method called `setLastTickMarkStyleProperty`, similar to `setThumbStyleProperty`:
```
  adjustLastTickMark(numMarks) {
    const lastStepRatio = (this.max_ - numMarks * this.step_) / this.step_ + 1;
    this.adapter_.setLastTickMarkStyleProperty('flex', String(lastStepRatio));
  }
```

Reply:
👍